apps/dotcdn/src/app/dotcdn.component.store.ts

```diff
@@ -0,0 +1,264 @@
+                                this.dispatchLoading({
+                                    loadingState: LoadingState.LOADED,
+                                    loader: Loader.CHART
+                                });
+
+                                const {
+                                    statsData,
+                                    chartData: [chartBandwidthData, chartRequestsData],
+                                    cdnDomain
+                                } = this.getChartStatsData(data);
+
+                                this.updateChartState({
+                                    chartBandwidthData,
+                                    chartRequestsData,
+                                    statsData,
+                                    cdnDomain
+                                });
+                            },
+                            (error) => {
+                                // TODO: Handle error
+                                console.log(error);
+                            }
+                        )
+                    );
+                })
+            );
+        }
+    );
+
+    /**
+     *  Dispatches a loading state
+     *
+     * @memberof DotCDNStore
+     */
+    readonly dispatchLoading = this.updater(
+        (state, action: { loadingState: string; loader: string }) => {
+            switch (action.loader) {
+                case Loader.CHART:
+                    return {
+                        ...state,
+                        isChartLoading: action.loadingState === LoadingState.LOADING
+                    };
+                case Loader.PURGE_URLS:
+                    return {
+                        ...state,
+                        isPurgeUrlsLoading: action.loadingState === LoadingState.LOADING
+                    };
+                case Loader.PURGE_PULL_ZONE:
+                    return {
+                        ...state,
+                        isPurgeZoneLoading: action.loadingState === LoadingState.LOADING
+                    };
+            }
+        }
+    );
+
+    /**
+     * Purges the CDN cache
+     *
+     * @param {boolean} [invalidateAll=false]
+     * @param {string[]} [urls]
+     * @return {*}  {(Observable<ResponseView<any>> | void)}
+     * @memberof DotCDNStore
+     */
+
+    purgeCDNCache(urls: string[]): Observable<PurgeReturnData> {
+        const loading$ = of(
+            this.dispatchLoading({
+                loadingState: LoadingState.LOADING,
+                loader: Loader.PURGE_URLS
+            })
+        );
+        return loading$.pipe(
+            switchMapTo(
+                this.dotCdnService.purgeCache(urls).pipe(
+                    tap(() => {
+                        this.dispatchLoading({
+                            loadingState: LoadingState.LOADED,
+                            loader: Loader.PURGE_URLS
+                        });
+                    })
+                )
+            )
+        );
+    }
```

Comment:
Why the `of` and why this is not an effect?

Reply:
Makes sense for it to be an effect. This was the initial impl as I was moving the method from the service to the store. Will re-impl.